R/get_phylopic.R

```diff
@@ -31,36 +36,50 @@ get_phylopic <- function(uuid = NULL, format = "vector") {
   if (!is.character(uuid)) {
     stop("`uuid` is not of class character.")
   }
+  if (as.character(format) %in% c("512", "1024", "1536")) {
+    lifecycle::deprecate_warn("1.1.0",
+                              paste0("get_phylopic(format = '",
+                                     "no longer supports values of ",
+                                     "512, 1024, or 1536')"),
+                              details = paste0("Use the `height` argument ",
+                                               "instead with the `format` ",
+                                               "argument set to \"raster\"."))
+    height <- as.numeric(format)
+    format <- "raster"
+  }
   format <- match.arg(as.character(format),
-                    c("64", "128", "192", "512", "1024", "1536", "twitter",
-                      "vector"))
+                      c("64", "128", "192", "raster", "twitter", "vector"))
   image_info <- phy_GET(file.path("images", uuid))$`_links`
+  ret <- NULL
   if (format %in% c("64", "128", "192")) { # get thumbnail url
     thumbs <- image_info$thumbnailFiles
     ind <- grepl(format, thumbs$sizes)
     if (!any(ind)) {
       ind <- 1
-      warning(paste("No raster image with dimension", format, "available.",
-                    "Returning raster image with dimensions", rasters$sizes[1],
-                    "instead."))
+      warning(paste("No thumbnail image with dimension", format, "available.",
+                    "Returning thumbnail image with dimensions",
+                    rasters$sizes[1], "instead."))
     }
     url <- thumbs$href[ind]
-  } else if (format %in% c("512", "1024", "1536")) { # get raster url
+  } else if (format == "raster") {
     rasters <- image_info$rasterFiles
-    ind <- grepl(format, rasters$sizes)
-    if (!any(ind)) {
-      ind <- 1
-      warning(paste("No raster image with dimension", format, "available.",
-                    "Returning raster image with dimensions", rasters$sizes[1],
-                    "instead."))
+    # check if there is an existing file with the desired height
+    ind <- grepl(paste0("x", height), rasters$sizes)
```

Comment:
Is there a big tradeoff in checking whether images are already available before downloading, rather than just getting the svg and creating the desired image size?

Reply:
I just checked this, and it's not a huge time save, but it does take more than twice as long to render the image as it does to just download it, so if you pick one of the magic numbers, it does save you some time.
```r
microbenchmark("download" = get_phylopic("a83f876a-0a00-4e21-898e-6e65816f40a6", format = "raster", height = 276),
                "render" = get_phylopic("a83f876a-0a00-4e21-898e-6e65816f40a6", format = "raster", height = 277),
                times = 100)
```
![image](https://user-images.githubusercontent.com/7232514/234019993-2cce98ac-7e27-40fa-961b-e925010d99a6.png)

Reply:
OK! Thanks for checking!